src/app/routes/search-result/search-result.component.ts

```diff
@@ -57,29 +55,39 @@ export class SearchResultComponent implements OnInit {
     });
     this.filteredOptions = new BehaviorSubject([]);
     this.searchForm.valueChanges
-      .pipe(distinctUntilChanged((a: any, b: any) => a.searchTxt === b.searchTxt && a.filterType === b.filterType))
-      .subscribe((value: any) => {
-        this.filteredOptions.next(this._filterOrgsAndApps(value.searchTxt, value.filterType));
+      .pipe(
+        distinctUntilChanged(
+          (a: any, b: any) =>
+            a.searchTxt === b.searchTxt && a.filterType === b.filterType
+        )
+      )
+      .subscribe(async (value: any) => {
+        const options = await this._filterOrgsAndApps(
+          value.searchTxt,
+          value.filterType
+        );
+        this.filteredOptions.next(options);
       });
 
     this.activeRoute.queryParams.subscribe(async (queryParams: any) => {
       // Get requested claims
-      this.requestedClaims = await this.iamService.iam.getRequestedClaims({
-        did: this.iamService.iam.getDid()
-      });
-      
-      // Retrieve List - Orgs & Apps
-      await this._getOrgsAndApps();
-
-      if (queryParams.keyword) {
-        this.searchForm.get('searchTxt').setValue(queryParams.keyword);
-      }
+      try {
+        this.requestedClaims = await this.iamService.iam.getRequestedClaims({
+          did: this.iamService.iam.getDid()
+        });
 
-      if (queryParams.namespace) {
-        this._initView(queryParams.namespace);
+        if (queryParams.keyword) {
+          this.searchForm.get('searchTxt').setValue(queryParams.keyword);
+        }
+
+        if (queryParams.namespace) {
+          await this._initView(queryParams.namespace);
+        }
+        this._manageScreenWidth();
+        this.loadingService.hide();
```

Comment:
Please put this in finally clause to make sure that the loading screen will be hidden regardless of success or failure of the try statement.

Reply:
👍  done